EdgeDeflector/resources/nsis_installer.nsi

```diff
@@ -1,72 +1,237 @@
 ; Installer for EdgeDeflector
 
+
+;--------------------------------
+;Include Modern UI
+
+  !include "MUI2.nsh"
+
 ;--------------------------------
+;General
+
+  ;Define name of the product
+  !define PRODUCT "EdgeDeflector"
+  !define ABOUTURL "https://github.com/da2x/EdgeDeflector"
+  !define INSTALLATIONURL "https://github.com/da2x/EdgeDeflector/wiki/Thank-you-for-installing-EdgeDeflector"
+
+  ;Properly display all languages
+  Unicode true
+
+  ;Show 'console' in installer and uninstaller
+  ShowInstDetails "show"
+  ShowUninstDetails "show"
+
+  ;Name and file
+  Name "${PRODUCT}"
+  Icon "..\Icon.ico"
+  LicenseData "..\..\LICENSE"
+  OutFile "..\bin\Release\${PRODUCT}_install.exe"
 
-!define PRODUCT "EdgeDeflector"
+  ;Version
+  !define VERSIONMAJOR 1
+  !define VERSIONMINOR 1
+  !define VERSIONBUILD 1
 
-Name "${PRODUCT} Installer"
+  ;Default installation folder
+  InstallDir "$PROGRAMFILES\${PRODUCT}"
 
-OutFile "..\bin\Release\${PRODUCT}_install.exe"
+  ;Get installation folder from registry if available
+  InstallDirRegKey HKLM "Software\${PRODUCT}" ""
 
-; Default installation directory
-InstallDir $PROGRAMFILES\${PRODUCT}
+  ;Request application admin
+  RequestExecutionLevel admin
 
-; Store install dir in the registry
-InstallDirRegKey HKLM "Software\${PRODUCT}" "Install_Dir"
+;--------------------------------
+;Interface Settings
+
+  ;Show warning if user wants to abort
+  !define MUI_ABORTWARNING
 
-; Request application privileges for UAC
-RequestExecutionLevel admin
+  ;Show all languages, despite user's codepage
+  !define MUI_LANGDLL_ALLLANGUAGES
 
+  ;Use the custom own icon
+  !define MUI_ICON "..\Icon.ico"
+  !define MUI_UNICON "UnIcon.ico"
+  !define MUI_HEADERIMAGE_RIGHT
+  !define MUI_WELCOMEFINISHPAGE_BITMAP "installer_right.bmp"
+  !define MUI_UNWELCOMEFINISHPAGE_BITMAP "uninstaller_right.bmp"
 
 ;--------------------------------
+;Language Selection Dialog Settings
 
-; Installer pages
-Page directory
-Page instfiles
+  ;Remember the installer language
+  !define MUI_LANGDLL_REGISTRY_ROOT "HKCU" 
+  !define MUI_LANGDLL_REGISTRY_KEY "Software\${PRODUCT}" 
+  !define MUI_LANGDLL_REGISTRY_VALUENAME "Installer Language"
+
+;--------------------------------
+;Pages
+
+  ;For the installer
+  !insertmacro MUI_PAGE_WELCOME
+  !insertmacro MUI_PAGE_LICENSE "..\..\LICENSE"
+  #!insertmacro MUI_PAGE_COMPONENTS
+  #!insertmacro MUI_PAGE_DIRECTORY
+  !insertmacro MUI_PAGE_INSTFILES
+  !insertmacro MUI_PAGE_FINISH
+
+  ;For the uninstaller
+  !insertmacro MUI_UNPAGE_WELCOME
+  !insertmacro MUI_UNPAGE_CONFIRM
+  !insertmacro MUI_UNPAGE_INSTFILES
+  !insertmacro MUI_UNPAGE_FINISH
 
-; Uninstaller pages
-UninstPage uninstConfirm
-UninstPage instfiles
+;--------------------------------
+;Languages
+
+  ;the first language is the default language
```

Comment:
What is the purpose of including this list? The project doesn’t have any translations for these languages.

Translating the installer would give people the false expectation that they can get support in their native languages. I only speak three, so this would be difficult to accommodate.

As for default language, shouldn’t that be the system locale and not whatever is at the top of this list?

Reply:
I added it because the installer get's automatically translated into any of these languages and so (besides the license and the window that pop's up at the end) everybody could install the program while using his mother tongue.
*(I did this also because using the program does't require to know English.)*
But if you only want to have the installer in English I can simply remove the whole thing. 

I'll still look up if you could use the System.Locale somehow - that's a good idea.

Reply:
I’d have to translate the welcome page (which opens to trigger the App selector dialog so people can choose EdgeDeflector instead of Microsoft Edge) also, right? I do believe people need to understand English at this time.

Reply:
Fixed this in the newest version - the language will automatically fetch the current language and search for it in the list of added languages (no dialog and save to registry any more) - if the language isn't in the list the first entry (English) will be chosen.